docs/components/tuner/callbacks.md

```diff
@@ -0,0 +1,67 @@
+# Callbacks
+
+Callbacks offer a way to integrate various auxiliary tasks into the training loop. We already offer built-in callbacks for some common tasks, such as
+- Showing a progress bar (which is shown by default)
+- [Tracking experiements](#experiement-tracking)
+
+You can also [write your own callbacks](#custom-callbacks).
+
+## Experiement Tracking
+
+When you train a model, you want to monitor the progress of the training. If you don't want to stay glued to the screen watching the training or evaluation loss change, you need to use an experiement tracking tool.
+
+Using such a tool also brings a lot of other benefits, such as the ability to easily compare different experiements.
+
+We currently offer integration with [Weights and Biases](https://wandb.ai/site) through the {class}`~finetuner.tuner.callback.wandb_logger.WandBLogger`. To use it, you must first download the `wandb` client and log into your account, which you can do with
+
+```
+pip install wandb
+wandb login
+```
+
+Then, simply create the callback instance and pass it to the `callbacks` argument of the Tuner
+
+```python
+from finetuner.tuner.callback import WandBLogger
+from finetuner.tuner.pytorch import PytorchTuner
+
+logger = WandBLogger()
+
+tuner = PytorchTuner(..., callbacks=[logger])
+```
+
+You should then be able to see your training runs in wandb.
+
+## Custom callbacks
+
+If the existing callbacks don't provide the functionality you need, you can easily write your own.
+
+To do that, you should subclass {class}`~finetuner.tuner.callback.base.BaseCallback`, and override the `on_*` methods that you need. Each method will get a `tuner` object as an argument - this gives you access to its `.state` attribute, which is a {class}`~finetuner.tuner.state.TunerState` object, and contains useful information on the current training run.
+
+Here's a simple example of a callback that time each training epoch, and prints the elapsed time at the end
```

Comment:
time`s` each training epoch.

Reply:
Full stop is missing in the end.

Reply:
@gvondulong for such changes next time use the inline edit tool - it will be easier for me to commit them

Reply:
![Screenshot from 2021-12-16 11-51-33](https://user-images.githubusercontent.com/11489772/146339089-011ff22a-e581-473b-89fc-c9e9d30bccb9.png)